lib/Immutable.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+const util = require('util');
+
+/**
+ * Error for Immutable variables
+ *
+ * @extends Error
+ * @class Errors.Immutable
+ * */
+class Immutable extends Error{
+    /**
+ * @example
+
+      let piKindOf = 22/7;
+
+      function getItRight(){
+          if(typeof piKindOf != 'undefined'){
+              const err = new Errors.Immutable;
+              err.setMessage(
+                  'piKindOf',
+                  'global'
+              );
+              throw err;
+          }
+          piKindOf = Math.PI;
+      }
+
+      getItRight();
+
```

Comment:
I think you should change this from being an example of a global object. perhaps to a value on a global object LIKE a={} a.b <-thats undefined, then you set it then you try to set it again. But use a getter and setter. ***Like a class***. The setter should throw this if the value is already set.

Global variable names are usuially defined with some value so this example isn't as clear as it could be.  It's a confusing example for noobs.